### DISCUSSION TITLE
Level of Detail (LOD) support?

### DISCUSSION THREAD

**Discussion by gravytates** [google/model-viewer#2774] [Q&A] [answered]
Hello! Are there any plans for adding LOD support for the model viewer? Or are there already known techniques for effectively achieving this (ie swapping model src based on camera distance)? I haven't seen it referenced in any docs or discussions yet.
Thanks!

**Comment by elalish** [comment, marked as answer]
The idea is  this is achievable today  by swapping the `src` as you suggested. I'd expect something like choosing the `src` to begin with by checking the element  resolution and  device type. Or perhaps loading a low-res model first and then pulling in the higher  res one  after if needed. If users start doing this a lot and there's an API we could provide to make it easier, we'll be  happy to.

**Reply by elalish** [reply]
related:  https://github.com/google/model-viewer/issues/429#issuecomment-917103247

**Reply by gravytates** [reply]
wonderful, thank you!
